src/sage/plot/misc.py

```diff
@@ -136,6 +136,14 @@ def setup_for_eval_on_grid(funcs,
         ranges = [r[1:] for r in ranges]
         if len(set(vars)) < len(vars):
             raise ValueError("range variables should be distinct, but there are duplicates")
+    
+    #to check if range is correctly defined by user. gives error is xmin>xmax or ymin>ymax
+    if (ranges[0][-2]>ranges[0][-1]):
+        raise ValueError("Range not correctly defined. xmin>xmax")
```

Comment:
I suggest the following error message:
"xrange not correctly defined: xmin(={}) > xmax(={})".format(ranges[0][-2], ranges[0][-1])